edit-post/api/plugin.js

```diff
@@ -0,0 +1,87 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
+
+/* External dependencies */
+import { isFunction, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { PluginContextProvider, withPluginContext } from './components/context';
+
+const plugins = {};
+
+/**
+ * Registers a plugin to the editor.
+ *
+ * @param {Object}   settings        The settings for this plugin.
+ * @param {string}   settings.name   The name of the plugin.
+ * @param {Function} settings.render The function that renders the plugin.
+ *
+ * @return {Object} The final plugin settings object.
+ */
+function registerPlugin( settings ) {
```

Comment:
Why is `registerPlugin` in `edit-post`? Will this pattern or the concept of plugin UI extensibility not apply to other screens?

Reply:
See: https://github.com/WordPress/gutenberg/pull/5430#issuecomment-372607425. It makes perfect sense to introduce new `plugins` module to align with `blocks`.